### PR TITLE
feat: spaces - resource list

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #2914 #2915 #2917 #3230

### Description:
- [x] Display them in table format, where the user can sort by name and time/last modified. We may need to experiment with column widths a bit.
- [x] Add chart / dashboard icons
- [x] Remove the chart/dashboard description icon from these components. Instead, show the descriptions on hover of the component.
- [x] Adds count
- [x] Updates Action menu styles
- [x] Update description from dashboard update
- [x] Normalize empty states

implemented new ResourceList component

CleanShot 2022-09-12 at 17 05 26@2x

sorting:
CleanShot 2022-09-12 at 17 02 51

action menu style updates:
CleanShot 2022-09-12 at 17 02 29@2x

empty states:
CleanShot 2022-09-12 at 17 02 23@2x

description tooltip
CleanShot 2022-09-12 at 17 07 28@2x

### CODE REVIEW

**Comment by IrakliJani** [comment]
@PriPatel want a quick confirmation from you (just have a look to the screenshots)

**Comment by PriPatel** [comment]
Great -  @IrakliJani , it looks like it works well :)

**Review by rephus — APPROVED** [review]
Looks good :+1:

File: packages/common/.eslintrc.json [hunk 1]
```diff
@@ -12,6 +12,11 @@
     },
     "rules": {
         "no-case-declarations": "off",
-        "no-template-curly-in-string": "off"
+        "no-template-curly-in-string": "off",
+        "no-unused-vars": ["warn", { "argsIgnorePattern": "^_" }],
+        "@typescript-eslint/no-unused-vars": [
+            "warn",
+            { "argsIgnorePattern": "^_" }
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/common/.eslintrc.json:19]: supposedly for unused variables which start with _ (_arg). I think I can get rid of that. if postfix _ `arg_` is being ignored by default

**Comment by rephus** [comment]
Tests are failing (unused var) and E2E test need to be updated to click on the new buttons (delete dashboard?)  BUt overall looks good when I tested it on render

**Comment by IrakliJani** [comment]
some UI touch ups

CleanShot 2022-09-15 at 19 48 21@2x

**Comment by IrakliJani** [comment]
@rephus

> Tests are failing (unused var) and E2E test need to be updated to click on the new buttons (delete dashboard?) But overall looks good when I tested it on render

Thanks for the review! just updated tests and I have experienced one snag. when I try to click a button this happens https://stackoverflow.com/questions/67391770/why-is-cypress-saying-my-element-is-detached-after-just-running-a-get-command
have you experienced this before?
my hunch is that it must be related to react-table
only `click({ force: true })` solved it

**Review by rephus — APPROVED** [review]

**Comment by rephus** [comment]
> Thanks for the review! just updated tests and I have experienced one snag. when I try to click a button this happens https://stackoverflow.com/questions/67391770/why-is-cypress-saying-my-element-is-detached-after-just-running-a-get-command have you experienced this before? my hunch is that it must be related to react-table only `click({ force: true })` solved it

Yes, I think I have seen it before. Feel free to force it if that fixes the issue

**Comment by IrakliJani** [comment]
@rephus I spend some time yesterday figuring out what was going on
tanstack/react-table goes beyond react to optimize it's rendering (also it's a framework agnostic library) and remounts dom without vdom (everything that is being rendered with `flexRender`): https://github.com/lightdash/lightdash/blob/7ec7c764c0457e53b8dce160a21200c104a16b70/packages/frontend/src/components/common/ResourceList/ResourceTable/index.tsx#L259-L262

so when the page loads, components that are rendered by column helpers are mounting/unmounting few times: https://github.com/lightdash/lightdash/blob/7ec7c764c0457e53b8dce160a21200c104a16b70/packages/frontend/src/components/common/ResourceList/ResourceTable/index.tsx#L145-L159

thus, nothing helps from here https://stackoverflow.com/questions/67391770/why-is-cypress-saying-my-element-is-detached-after-just-running-a-get-command, including `click({ force: true })`

putting `.wait(1000)` helps (as after some time components stop unmounting) but this is a very bad hack and will not work for highly dynamic and interactive tables.

cc: @lightdash/engineering

**Comment by rephus** [comment]
mmmm, ok, let's try this for now, as long as it does not fails randomly I'm fine with it. Well done.

**Comment by ZeRego** [comment]
> putting .wait(1000) helps (as after some time components stop unmounting) but this is a very bad hack and will not work for highly dynamic and interactive tables.

@IrakliJani It is a bad practice but not a blocker right now. Let's ship it and we see how it behaves 👍

**Comment by ZeRego** [comment]
@all-contributors add @IrakliJani for code, bug, design, tests

**Pull request merged**